pallets/lbp/src/lib.rs

```diff
@@ -0,0 +1,987 @@
+				asset_a,
+				asset_b,
+				sale_duration,
+				weight_curve,
+				pausable,
+			);
+			Self::validate_pool_data(&pool_data)?;
+
+			let deposit = T::PoolDeposit::get();
+
+			T::MultiCurrency::reserve(T::NativeAssetId::get(), &pool_owner, deposit)?;
+			let pool_id = Self::get_pair_id(asset_pair);
+			<PoolDeposit<T>>::insert(&pool_id, &deposit);
+
+			<PoolData<T>>::insert(&pool_id, &pool_data);
+
+			T::MultiCurrency::transfer(asset_a.id, &pool_owner, &pool_id, asset_a.amount)?;
+			T::MultiCurrency::transfer(asset_b.id, &pool_owner, &pool_id, asset_b.amount)?;
+
+			Self::deposit_event(Event::PoolCreated(
+				pool_owner,
+				pool_id,
+				asset_a.id,
+				asset_b.id,
+				asset_a.amount,
+				asset_b.amount,
+			));
+
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::update_pool_data())]
+		#[transactional]
+		pub fn update_pool_data(
+			origin: OriginFor<T>,
+			pool_id: PoolId<T>,
+			start: Option<T::BlockNumber>,
+			end: Option<T::BlockNumber>,
+			initial_weights: Option<((AssetId, LBPWeight), (AssetId, LBPWeight))>,
+			final_weights: Option<((AssetId, LBPWeight), (AssetId, LBPWeight))>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			<PoolData<T>>::try_mutate_exists(pool_id.clone(), |maybe_pool| -> DispatchResultWithPostInfo {
+				// check existence of the pool
+				let mut pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+
+				ensure!(
+					start.is_some() || end.is_some() || initial_weights.is_some() || final_weights.is_some(),
+					Error::<T>::NothingToUpdate
+				);
+
+				Self::ensure_pool_ownership(&who, &pool_id)?;
+
+				let now = <frame_system::Pallet<T>>::block_number();
+
+				ensure!(pool.start == Zero::zero() || now < pool.start, Error::<T>::SaleStarted);
+
+				pool.start = start.unwrap_or(pool.start);
+				pool.end = end.unwrap_or(pool.end);
+				if let Some(initial_weights) = initial_weights {
+					pool.initial_weights = Self::get_weights_in_order(pool, initial_weights)?;
+				}
+				if let Some(final_weights) = final_weights {
+					pool.final_weights = Self::get_weights_in_order(pool, final_weights)?;
+				}
+
+				Self::validate_pool_data(&pool)?;
+
+				Self::deposit_event(Event::PoolUpdated(who, pool_id));
+				Ok(().into())
+			})
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::pause_pool())]
+		pub fn pause_pool(origin: OriginFor<T>, pool_id: PoolId<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			<PoolData<T>>::try_mutate_exists(pool_id.clone(), |maybe_pool| -> DispatchResultWithPostInfo {
+				// check existence of the pool
+				let mut pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+
+				Self::ensure_pool_ownership(&who, &pool_id)?;
+
+				ensure!(pool.pausable, Error::<T>::PoolIsNotPausable);
+				ensure!(!pool.paused, Error::<T>::CannotPausePausedPool);
+
+				let now = <frame_system::Pallet<T>>::block_number();
+				ensure!(pool.end > now, Error::<T>::CannotPauseEndedPool);
```

Comment:
as mentioned - perhaps a little helper for this would be useful. 

And if feel it should be `>='.